test/dataset_test.cpp

```diff
@@ -574,14 +574,20 @@ TEST(Dataset, operator_multiplication_broadcast) {
   Dataset a;
   a.insert(Data::Value, Dimensions({{Dim::X, 2}, {Dim::Y, 2}}),
            {1.0, 1.0, 1.0, 1.0});
+  a(Data::Value).setUnit(units::m);
   a.insert(Data::Variance, Dimensions({{Dim::X, 2}, {Dim::Y, 2}}),
            {1.0, 1.0, 1.0, 1.0});
 
   Dataset b;
   b.insert(Data::Value, {Dim::Y, 2}, {2.0, 3.0});
+  b(Data::Value).setUnit(units::m);
   b.insert(Data::Variance, {Dim::Y, 2}, {1.0, 1.0});
 
   auto c = a * b;
+  // Check units
```

Comment:
Does it work with `a / a` ? We were having problems with dimensionless not being the same as `si::dimensionless`

Reply:
@nvaytet I've just added a new test. S

Reply:
Works for `c = a / b` too.

Reply:
Canout you also try with `units.counts`? Because we were having problems with `counts^2 * counts^2`

Reply:
Was your problem specifically around examples where there is dimension mismatch between rhs and lhs arguments? The changes here relate only to broadcast - I'm happy to check though.

Reply:
So extending the unit definitions to include `counts ^ 4` and switching units to `counts` ensures that both these new test methods pass. However, note this is specifically tested against for failure [here](https://github.com/mantidproject/dataset/blob/master/test/dataset_test.cpp#L836), which makes sense